common/OpTestHost.py

```diff
@@ -918,7 +918,7 @@ def host_get_list_of_chips(self):
         return chips
 
     def host_get_cores(self):
-        proc_gen = ''.join(self.host_run_command("grep '^cpu' /proc/cpuinfo |uniq|sed -e 's/^.*: //;s/ .*//;'"))
+        proc_gen = self.host_get_proc_gen()
```

Comment:
So you are doing multiple things in single patch. Ideally all these helper function should go as separate patch.

Also when you create such helper function try to fix all existing code (ex: testcases/OpTestPrdDriver.py should make sure of new helper function to get proc_generation).

Reply:
Okay, will take care of it.